packages/language-server/src/completion-items.ts

```diff
@@ -454,10 +475,24 @@ function getAddNamespaceEdit(
   // xml document will not be empty
   /* istanbul ignore else */
   if (parent.rootElement !== null) {
-    const position =
-      parent.rootElement.syntax.openName ?? parent.rootElement.position;
-    // We want to insert, not replace - the actual position should be at the end of the range.
-    const range = createInsertRange(position.endLine, position.endColumn);
+    let range: Range;
+    if (parent.rootElement.syntax.openName !== undefined) {
+      const position = parent.rootElement.syntax.openName;
+      // We want to insert the namespace at the end of the range, after the tag name.
+      range = createInsertRangeAfter(position.endLine, position.endColumn);
+    } else {
+      const position =
+        parent.rootElement.syntax.openBody ?? parent.rootElement.position;
```

Comment:
TODO: consider change XML-Tools/AST and adding the "<" token to the syntax.
This will reduce a branch in the code, and avoid the need for 4 lines comment.

Perhaps add it as a TODO / do it in another PR (later) due to time constraints ...